llvm/include/llvm/IR/PatternMatch.h

```diff
@@ -433,7 +433,7 @@ template <typename Predicate> struct apf_pred_ty : public Predicate {
     if (V->getType()->isVectorTy())
       if (const auto *C = dyn_cast<Constant>(V))
         if (auto *CI = dyn_cast_or_null<ConstantFP>(
-                C->getSplatValue(/* AllowUndef */ true)))
+                C->getSplatValue(/* AllowPoison */ true)))
```

Comment:
Should we default the corresponding `api_pred_ty` to true?

Reply:
I think so, but not as part of this change.

Reply:
I've been looking into this. From a very quick inspection of all relevant matchers, I think only the usage in foldSelectICmpAndBinOp is problematic if poison is allowed.

However, our test coverage seems to be pretty bad -- allowing poison in api_pred_ty only results in test changes for a single transform :( At the same time, I don't really want to spend time adding poison vector tests for dozens of affected transforms...

Reply:
The usage in foldSelectICmpAndBinOp is fine after all. I submitted https://github.com/llvm/llvm-project/pull/89188 to get the discussion going on this, not sure whether our current test coverage will be a blocker for this or not.